skyproj/skyproj.py

```diff
@@ -101,8 +102,25 @@ def __init__(self, ax=None, projection_name='cyl', lon_0=0, gridlines=True, cele
         self._wrap = (lon_0 + 180.) % 360.
         self._lon_0 = self.projection.proj4_params['lon_0']
 
+        self._full_sky = False
+        self._full_sky_extent = [
+            lon_0 - 180.0,
+            lon_0 + 180.0,
+            -90.0 + self._pole_clip,
+            90.0 - self._pole_clip
+        ]
+
+        self._full_sky = False
         if extent is None:
-            extent = [lon_0 - 180.0, lon_0 + 180.0, -90.0 + self.pole_clip, 90.0 - self.pole_clip]
+            extent = self._full_sky_extent
+            self._full_sky = True
+        elif np.allclose(extent, self._full_sky_extent):
```

Comment:
Rather than `allclose`, should this be some kind of comparison where you ask if the extents are >= the default `full_sky_extent`? I'm not sure that this is exactly where the code is failing, but I noticed that if I start with a full_sky map, then zoom into a specific region (so `self._full_sky` gets set to `False`), then zoom back out to an extent where you see the full sky map,  then  `self._full_sky` is still set to `False` and the frame is still shown.

Reply:
Following up on this, for the McBryde map
```
In [63]: smap._full_sky_extent
Out[63]: 
(179.94709441501143,
 -179.94709441501143,
 -89.83508654926219,
 89.83508654926219)
```
but once I've zoomed out
```
In [62]: smap._ax.get_extent(lonlat=True)
Out[62]: (179.99999, -179.99999, -89.99999, 89.99999)
```
and 
```
In [66]: np.allclose(smap._ax.get_extent(lonlat=True), smap._full_sky_extent)
Out[66]: False
```

Reply:
I thought I fixed that problem.  Argh.  Can you tell me your exact steps that led to this?  

But the true solution might be to have it do no frame, only do the boundaries as necessary, curved or not.

Reply:
```
smap = skyproj.McBrydeSkyproj()
smap.scatter([0, 90], [0, 45])
print(smap._full_sky, smap._full_sky_extent)
>>> True (179.94709441501143, -179.94709441501143, -89.83508654926219, 89.83508654926219)
```
Zoom into a subregion of the map (click "Zoom to rectangle" tool and draw a box)
```
print(smap._full_sky, smap._ax.get_extent())
>>> False (65.51432334530101, -70.05211197527862, -42.92936235009513, 24.73023811486473)
```
Then interactively zoom back out to view the full sky (select the "Pan/Zoom" tool, hold right click and drag mouse down)...
```
print(smap._full_sky, smap._ax.get_extent())
>>> False (179.99999, -179.99999, -89.99999, 89.99999)
```
![image](https://user-images.githubusercontent.com/5605123/155772306-d0812681-d8d3-4e4c-af3f-c28cced419c2.png)